# Assignment: Student Wellbeing Statistical Analysis & Probability (Day 16)

Using the provided Student Wellbeing Survey dataset, create one Jupyter/Google Colab notebook and complete the following analysis precisely. First, inspect the dataset and calculate the mean, median, and mode for Weekly_Study_Hours, Average_Sleep_Hours, Daily_Screen_Time_Hours, Stress_Score, and Academic_Readiness_Score, then calculate the range, variance, standard deviation, Q1, Q3, and IQR for the same variables and identify which variable has the greatest variability. Use the IQR method to identify outliers in Weekly_Study_Hours, Daily_Screen_Time_Hours, Commute_Time_Minutes, and Monthly_Discretionary_Spending, and compare the mean and median of at least one variable before and after removing its detected outliers. For probability, define Event A = student has a Part_Time_Job of "Yes", Event B = student has Stress_Score ≥ 7, Event C = student has Scholarship = "Yes", and Event D = student exercises at least 3 days per week; calculate P(A), P(B), P(C), P(D), P(A or B), P(A and B), P(A|B), and P(B|A), and determine whether Year_of_Study = 1 and Year_of_Study = 4 are mutually exclusive events. Investigate whether A and B appear independent by comparing P(A and B) with P(A) × P(B), and explain your conclusion. Then use Bayes' theorem to calculate P(A|B) using P(B|A), P(B|not A), and P(A), and verify that your answer agrees with the direct conditional-probability calculation. Finally, use Academic_Readiness_Score to study the normal distribution by calculating its mean and standard deviation, finding the Z-score of the student with the highest score and the student with the lowest score, interpreting both Z-scores, and using the 68-95-99.7 empirical rule to determine the expected percentage of students within 1, 2, and 3 standard deviations of the mean. Throughout the notebook, show the relevant formula, Python calculation, numerical answer, and a brief interpretation, and conclude with 5 meaningful statistical observations based on your results.


## 1. Dataset Inspection
load data set

In [1]:
import pandas as pd
import numpy as np

data = pd.read_csv("Day16_Student_Wellbeing_Survey.csv")

print("Shape -> Rows :", data.shape[0], "| Columns :", data.shape[1])
print()
print("First 3 rows:")
print(data.head(3).to_string())
print()
print("Column Data Types:")
print(data.dtypes.to_string())

Shape -> Rows : 600 | Columns : 20

First 3 rows:
  Student_ID  Age        Faculty  Year_of_Study        City Accommodation Scholarship Part_Time_Job Internet_Quality Preferred_Study_Space  Weekly_Study_Hours  Average_Sleep_Hours  Daily_Screen_Time_Hours  Exercise_Days_Per_Week  Commute_Time_Minutes  Stress_Score  Academic_Readiness_Score  Overall_Satisfaction  Monthly_Discretionary_Spending  Social_Activity_Hours_Per_Week
0    STU0001   21   Data Science              3   Bengaluru        Hostel          No           Yes             Good                  Café                15.2                  5.7                      3.2                       3                  11.8           5.4                      63.1                   3.2                          5131.0                             7.8
1    STU0002   21   Data Science              4        Pune        Hostel         Yes            No             Poor               Library                17.6                  7.7                 

## 2. Mean, Median, Mode

**Mean** = sum / count , **Median** = middle value , **Mode** = most frequent value

In [2]:
cols = ["Weekly_Study_Hours",
        "Average_Sleep_Hours",
        "Daily_Screen_Time_Hours",
        "Stress_Score",
        "Academic_Readiness_Score"]

for col in cols:
    print(col)
    print("Mean--> Average value:", round(data[col].mean(), 2))
    print("Median--> Middle value:", round(data[col].median(), 2))
    print("Mode--> most frequent value:", data[col].mode()[0])
    print()

Weekly_Study_Hours
Mean--> Average value: 15.69
Median--> Middle value: 15.4
Mode--> most frequent value: 13.1

Average_Sleep_Hours
Mean--> Average value: 7.0
Median--> Middle value: 7.0
Mode--> most frequent value: 7.0

Daily_Screen_Time_Hours
Mean--> Average value: 4.5
Median--> Middle value: 4.2
Mode--> most frequent value: 3.5

Stress_Score
Mean--> Average value: 4.46
Median--> Middle value: 4.5
Mode--> most frequent value: 4.7

Academic_Readiness_Score
Mean--> Average value: 71.77
Median--> Middle value: 71.65
Mode--> most frequent value: 71.1



## 3. Dispersion - Range, Variance, Std Dev, Q1, Q3, IQR

**Range** = max - min , **Variance** = spread from mean , **Std Dev** = sqrt(variance) , **IQR** = Q3 - Q1

In [3]:
for col in cols:
    q1 = data[col].quantile(0.25)
    q3 = data[col].quantile(0.75)
    iqr = q3 - q1

    print(col)
    print("Range--> ", round(data[col].max() - data[col].min(), 2))
    print("Variance--> ", round(data[col].var(), 2))
    print("Std Dev--> ", round(data[col].std(), 2))
    print("Q1--> ", round(q1, 2), "| Q3--> ", round(q3, 2), "| IQR--> ", round(iqr, 2))
    print()

most_sd = max(round(data[c].std(), 2) for c in cols)
for c in cols:
    if round(data[c].std(), 2) == most_sd:
        print("Variable with greatest variability (highest Std Dev):", c, "| SD =", most_sd)

Weekly_Study_Hours
Range-->  30.0
Variance-->  19.22
Std Dev-->  4.38
Q1-->  13.0 | Q3-->  18.42 | IQR-->  5.42

Average_Sleep_Hours
Range-->  5.0
Variance-->  0.7
Std Dev-->  0.84
Q1-->  6.48 | Q3-->  7.6 | IQR-->  1.12

Daily_Screen_Time_Hours
Range-->  11.2
Variance-->  3.47
Std Dev-->  1.86
Q1-->  3.2 | Q3-->  5.4 | IQR-->  2.2

Stress_Score
Range-->  8.4
Variance-->  2.78
Std Dev-->  1.67
Q1-->  3.3 | Q3-->  5.7 | IQR-->  2.4

Academic_Readiness_Score
Range-->  56.1
Variance-->  93.98
Std Dev-->  9.69
Q1-->  65.2 | Q3-->  78.03 | IQR-->  12.83

Variable with greatest variability (highest Std Dev): Academic_Readiness_Score | SD = 9.69


## 4. Outlier Detection using IQR Method

**Lower Fence** = Q1 - 1.5*IQR , **Upper Fence** = Q3 + 1.5*IQR . Any value outside is an outlier.

In [4]:
outlier_cols = ["Weekly_Study_Hours",
                "Daily_Screen_Time_Hours",
                "Commute_Time_Minutes",
                "Monthly_Discretionary_Spending"]

for col in outlier_cols:
    q1 = data[col].quantile(0.25)
    q3 = data[col].quantile(0.75)
    iqr = q3 - q1
    lower = q1 - 1.5 * iqr
    upper = q3 + 1.5 * iqr
    outliers = data[(data[col] < lower) | (data[col] > upper)]

    print(col)
    print("Lower fence:", round(lower, 2))
    print("Upper fence:", round(upper, 2))
    print("Outliers:", len(outliers))
    print()

Weekly_Study_Hours
Lower fence: 4.86
Upper fence: 26.56
Outliers: 8

Daily_Screen_Time_Hours
Lower fence: -0.1
Upper fence: 8.7
Outliers: 18

Commute_Time_Minutes
Lower fence: -13.85
Upper fence: 58.75
Outliers: 4

Monthly_Discretionary_Spending
Lower fence: -1430.25
Upper fence: 13351.75
Outliers: 20



## 5. Mean vs Median - Before and After Removing Outliers

Removing outliers from Monthly_Discretionary_Spending to see the effect on mean vs median.

In [5]:
col = "Monthly_Discretionary_Spending"

q1 = data[col].quantile(0.25)
q3 = data[col].quantile(0.75)
iqr = q3 - q1
lower = q1 - 1.5 * iqr
upper = q3 + 1.5 * iqr

clean = data[(data[col] >= lower) & (data[col] <= upper)]

print("Rows removed:", len(data) - len(clean))
print()
print("Before -> Mean:", round(data[col].mean(), 2), "| Median:", round(data[col].median(), 2))
print("After  -> Mean:", round(clean[col].mean(), 2), "| Median:", round(clean[col].median(), 2))
print()
print("Mean changed a lot,median is more robust or stable")

Rows removed: 20

Before -> Mean: 6343.78 | Median: 5685.5
After  -> Mean: 5973.72 | Median: 5590.5

Mean changed a lot,median is more robust or stable


## 6. Probability

Event A = Part-time job (Yes) , B = Stress >= 7 , C = Scholarship (Yes) , D = Exercise >= 3 days

**P(Event)** = favourable / total

In [6]:
A = data["Part_Time_Job"] == "Yes"
B = data["Stress_Score"] >= 7
C = data["Scholarship"] == "Yes"
D = data["Exercise_Days_Per_Week"] >= 3
total = len(data)

Prob_A = A.sum() / total
Prob_B = B.sum() / total
Prob_C = C.sum() / total
Prob_D = D.sum() / total

print("Students -> A:", A.sum(), "| B:", B.sum(), "| C:", C.sum(), "| D:", D.sum())
print("P(A) =", round(Prob_A, 4))
print("P(B) =", round(Prob_B, 4))
print("P(C) =", round(Prob_C, 4))
print("P(D) =", round(Prob_D, 4))

Students -> A: 152 | B: 45 | C: 184 | D: 356
P(A) = 0.2533
P(B) = 0.075
P(C) = 0.3067
P(D) = 0.5933


In [7]:
Prob_A_and_B = (A & B).sum() / total
Prob_A_or_B = (A | B).sum() / total
Prob_A_given_B = Prob_A_and_B / Prob_B
Prob_B_given_A = Prob_A_and_B / Prob_A

print("P(A and B) =", round(Prob_A_and_B, 4))
print("P(A or B) =", round(Prob_A_or_B, 4))
print("P(A|B) =", round(Prob_A_given_B, 4))
print("P(B|A) =", round(Prob_B_given_A, 4))

P(A and B) = 0.0517
P(A or B) = 0.2767
P(A|B) = 0.6889
P(B|A) = 0.2039


In [8]:
year1 = data["Year_of_Study"] == 1
year4 = data["Year_of_Study"] == 4
both = (year1 & year4).sum()

print("Students in Year 1 AND Year 4 together:", both)
print("P(Year 1 and Year 4) =", round(both / total, 4))
print("Joint is 0 -> MUTUALLY EXCLUSIVE (can't be in both years)")
print()

product = Prob_A * Prob_B
print("Independence check:")
print("P(A and B)  =", round(Prob_A_and_B, 4))
print("P(A) x P(B) =", round(product, 4))
print("Not equal -> events A and B are DEPENDENT")

Students in Year 1 AND Year 4 together: 0
P(Year 1 and Year 4) = 0.0
Joint is 0 -> MUTUALLY EXCLUSIVE (can't be in both years)

Independence check:
P(A and B)  = 0.0517
P(A) x P(B) = 0.019
Not equal -> events A and B are DEPENDENT


## 7. Bayes' Theorem Verification

P(A|B) = [ P(B|A) x P(A) ] / [ P(B|A) x P(A) + P(B|not A) x P(not A) ]

In [9]:
not_A = ~A
P_not_A = not_A.sum() / total
P_B_given_notA = (B & not_A).sum() / not_A.sum()

numerator = Prob_B_given_A * Prob_A
denominator = numerator + (P_B_given_notA * P_not_A)
bayes = numerator / denominator

print("P(B|A) =", round(Prob_B_given_A, 4))
print("P(B|not A) =", round(P_B_given_notA, 4))
print("P(not A) =", round(P_not_A, 4))
print()
print("Bayes result  P(A|B):", round(bayes, 4))
print("Direct result P(A|B):", round(Prob_A_given_B, 4))
print("Both match -> Bayes' theorem VERIFIED")

P(B|A) = 0.2039
P(B|not A) = 0.0312
P(not A) = 0.7467

Bayes result  P(A|B): 0.6889
Direct result P(A|B): 0.6889
Both match -> Bayes' theorem VERIFIED


## 8. Normal Distribution & Z-Scores

**Z** = (X - Mean) / Std Dev . Rule: ~68% within 1 SD, 95% within 2 SD, 99.7% within 3 SD.

In [10]:
score = data["Academic_Readiness_Score"]
mu = score.mean()
sigma = score.std()

top = data.loc[score.idxmax()]
low = data.loc[score.idxmin()]

z_top = (top["Academic_Readiness_Score"] - mu) / sigma
z_low = (low["Academic_Readiness_Score"] - mu) / sigma

print("Mean =", round(mu, 2), "| Std Dev =", round(sigma, 2))
print("Top student:", top["Student_ID"], "score", top["Academic_Readiness_Score"], "Z =", round(z_top, 2))
print("Low student:", low["Student_ID"], "score", low["Academic_Readiness_Score"], "Z =", round(z_low, 2))
print()

w1 = ((score >= mu - sigma) & (score <= mu + sigma)).mean() * 100
w2 = ((score >= mu - 2*sigma) & (score <= mu + 2*sigma)).mean() * 100
w3 = ((score >= mu - 3*sigma) & (score <= mu + 3*sigma)).mean() * 100

print("Within 1 SD -> Expected 68% | Actual", round(w1, 2), "%")
print("Within 2 SD -> Expected 95% | Actual", round(w2, 2), "%")
print("Within 3 SD -> Expected 99.7% | Actual", round(w3, 2), "%")
print("Very close -> score is approximately NORMAL")
print()
print("Z-score interpretation:")
print("Top student Z =", round(z_top, 2), "-> about", str(round(z_top, 2)), "standard deviations ABOVE average (very good performance)")
print("Low student Z =", round(z_low, 2), "-> about", str(abs(round(z_low, 2))), "standard deviations BELOW average (extreme low, very rare)")

Mean = 71.77 | Std Dev = 9.69
Top student: STU0131 score 98.0 Z = 2.71
Low student: STU0543 score 41.9 Z = -3.08

Within 1 SD -> Expected 68% | Actual 69.5 %
Within 2 SD -> Expected 95% | Actual 95.17 %
Within 3 SD -> Expected 99.7% | Actual 99.83 %
Very close -> score is approximately NORMAL

Z-score interpretation:
Top student Z = 2.71 -> about 2.71 standard deviations ABOVE average (very good performance)
Low student Z = -3.08 -> about 3.08 standard deviations BELOW average (extreme low, very rare)


## Key Findings

1. **Sleep is most stable** - mean, median and mode of Average_Sleep_Hours all = 7.0, tiny Std Dev 0.84.
2. **Academic readiness varies most** - biggest spread (SD 9.69, Range 56.1).
3. **Outliers pull the mean, not the median** - removing 20 extreme spenders dropped mean 6343.78 -> 5973.72 but median barely moved (5685.50 -> 5590.50).
4. **Part-time job & high stress are dependent** - P(A and B) = 0.0517 != P(A) x P(B) = 0.019; P(A|B) = 0.6889 (Bayes verified same value). Year 1 & Year 4 are mutually exclusive (joint prob = 0).
5. **Readiness score is approx. normal** - actual coverage 69.5% / 95.17% / 99.83% matches the 68-95-99.7 rule. Topper Z = +2.71, lowest Z = -3.08.